data/processors/search.py

```diff
@@ -46,6 +46,16 @@ def add_ranking_base(data):
         else:
             ranking_factors["rank_usage"] = 100
 
+        # Type-specific boosts
+        stats = _data.get("props", {}).get("stats", None)
+        if stats is not None:
+            if _data["type"] == "room" and "n_seats" in stats:
+                ranking_factors["rank_boost"] = min(stats["n_seats"] // 10, 99)
+            if _data["type"] in {"building", "joined_building"} and "n_rooms_reg" in stats:
+                ranking_factors["rank_boost"] = min(stats["n_rooms_reg"] // 20, 99)
+            if _data["type"] in {"campus", "area", "site"} and "n_buildings" in stats:
+                ranking_factors["rank_boost"] = min(stats["n_buildings"], 99)
```

Comment:
Reasoning:
1. if stats: is equivalent, because
```
>>> bool(None)
False
>>> bool({})
False
>>> bool({2:3})
True
```
2. this looks a bit intimidating at first glance. I dont know, if this made it worse or better..
```suggestion
        if stats:
            rank_boost = None
            if _data["type"] == "room" and "n_seats" in stats:
                rank_boost = stats["n_seats"] // 10
            elif _data["type"] in {"building", "joined_building"} and "n_rooms_reg" in stats:
                rank_boost = stats["n_rooms_reg"] // 20
            elif _data["type"] in {"campus", "area", "site"} and "n_buildings" in stats:
                rank_boost = stats["n_buildings"]
            
            if rank_boost is not None:
                ranking_factors["rank_boost"] = min(rank_boost, 99)
```